### PR TITLE
test: Move CLI expect-tests to separate workflow; test Mac support there

### PR DESCRIPTION
**Pull request by timmc-edx**
A few test changes were required to account for slight differences in environments:
    
- Fix Ctrl-c test: Did not account for nondeterministic output ordering
- Fix "make interrupt" regex (not sure why this was working on Linux before!)

----

I've completed each of the following, or confirmed they do not apply to this PR:

- [x] Tested on both Mac and Linux (if changed Makefile or shell scripts)
- [x] Made a plan to communicate any major developer interface changes
    - Will announce on edx-internal #dev and openedx #general -- not that there are any interface changes, but this Mac/Docker/GH Actions integration was non-trivial and it would be good to spread the solution around a bit.

### CODE REVIEW

File: tests/metrics.py [hunk 1]
```diff
@@ -279,16 +279,18 @@ def test_handle_ctrl_c():
         # Make sure wrapped command has started before we interrupt,
         # otherwise signal handler won't even have been registered
         # yet.
-        p.expect(b'Are you sure you want to run this command')
+        p.expect(r'Are you sure you want to run this command')
```
Comment (timmc-edx) [inline tests/metrics.py:282]: Fixing an unrelated typo here. (No functional change, and could be a regular string, but using "raw strings" is a good reminder that these are in fact regexes.)

File: .github/workflows/cli-tests.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          # correct in the general case, but these Docker environment
+          # variables shouldn't contain newlines or escape sequences.
+          # They'll look like this:
+          #   DOCKER_TLS_VERIFY="1"
+          #   DOCKER_HOST="tcp://192.168.99.100:2376"
+          #   DOCKER_CERT_PATH="/Users/runner/.docker/machine/machines/default"
+          #   DOCKER_MACHINE_NAME="default"
+          #
+          # Docs on GITHUB_ENV:
+          # https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable
```
Comment (timmc-edx) [inline .github/workflows/cli-tests.yml]: I'm inclined to remove these lines from the comment.

File: .github/workflows/cli-tests.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
+name: CLI tests
```
Comment (robrap) [inline .github/workflows/cli-tests.yml:4]: I feel like I'd like a file-level comment, like a module docstring, to state the purpose of the workflow. Something to add a bit more color to what "cli-tests" is vs "ci".

File: .github/workflows/cli-tests.yml (hunk 2 above)
Comment (robrap) [reply .github/workflows/cli-tests.yml]: I think this comment is fine. It made it immediately clear to me that $GITHUB_ENV was a standard thing, and not something you are setting up as part of the workflows.

**Review by robrap — APPROVED** [review]
I think this is all good, but not certain and would trust better that it passes, and that it doesn't pass when broken (which I presume you have tested).

File: .github/workflows/cli-tests.yml (hunk 3 above)
Comment (timmc-edx) [reply .github/workflows/cli-tests.yml:4]: Yeah, that sounds good.

File: .github/workflows/cli-tests.yml (hunk 2 above)
Comment (timmc-edx) [reply .github/workflows/cli-tests.yml]: (I had highlighted lines 71 through 78 to make this comment, but it looks like the resulting lines-scope indicator is subtle. Oh well.)

Makes sense about GITHUB_ENV.

How about the Docker env vars? If I keep them I should either remove the quotes or add `export ` in front of each one in order to make it look like either the input or the output format...

Comment (robrap) [reply .github/workflows/cli-tests.yml]: I liked the large comment. Maybe include fewer variables (like 2) if you think it is getting too long, but I'd just make it a Before and After to make it clear on both sides.

**Comment by timmc-edx** [comment]
Incidentally, this ends up using boot2docker, which the README specifically says not to use. :man_facepalming: It seems to work for some Docker-requiring actions But I'm not sure how to install Mac Docker for Desktop in a Github Action.

**Comment by robrap** [comment]
Still approved as-is. But, does your comment https://github.com/edx/devstack/pull/761#issuecomment-853889581 warrant a comment in the code somewhere?

**Comment by timmc-edx** [comment]
Yeah, I'll add a comment about that and we can explore other options later if we want.

**Pull request merged**